vsintegration/tests/FSharp.Editor.Tests/Refactors/RefactorTestFramework.fs

```diff
@@ -0,0 +1,94 @@
+﻿module FSharp.Editor.Tests.Refactors.RefactorTestFramework
+
+open System
+open System.Linq
+open System.Collections.Immutable
+open System.Collections.Generic
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open Microsoft.VisualStudio.FSharp.Editor.CancellableTasks
+
+open FSharp.Editor.Tests.Helpers
+open Microsoft.CodeAnalysis.CodeRefactorings
+open Microsoft.CodeAnalysis.CodeActions
+open System.Threading
+open FSharp.Compiler.Text
+
+let GetTaskResult (task: Tasks.Task<'T>) = task.GetAwaiter().GetResult()
+
+type TestContext(Solution: Solution, CancellationToken) =
+    let mutable _solution = Solution
+    member this.CT = CancellationToken
+
+    member this.Solution
+        with set value = _solution <- value
+        and get () = _solution
+
+    interface IDisposable with
+        member this.Dispose() = Solution.Workspace.Dispose()
+
+    static member CreateWithCode(code: string) =
+        let solution = RoslynTestHelpers.CreateSolution(code)
+        let ct = CancellationToken false
+        new TestContext(solution, ct)
+
+    static member CreateWithCodeAndDependency (code: string) (codeForPreviousFile: string) =
+        let mutable solution = RoslynTestHelpers.CreateSolution(codeForPreviousFile)
+
+        let firstProject = solution.Projects.First()
+        solution <- solution.AddDocument(DocumentId.CreateNewId(firstProject.Id), "test2.fs", code, filePath = "C:\\test2.fs")
+
+        let ct = CancellationToken false
+        new TestContext(solution, ct)
+
+let mockAction =
+    Action<CodeActions.CodeAction, ImmutableArray<Diagnostic>>(fun _ _ -> ())
+
+let tryRefactor (code: string) (cursorPosition) (context: TestContext) (refactorProvider: 'T :> CodeRefactoringProvider) =
+    let refactoringActions = new List<CodeAction>()
```

Comment:
Maybe to simplify things a bit you can just have here this:
```
let mutable action : CodeAction = null
```
and then just assign it when ctoring `CodeRefactoringContext`.

Reply:
What do you mean? You're still using actions here so far :)